fiftyone/utils/ultralytics.py

```diff
@@ -282,7 +287,7 @@ def _to_keypoints(result, confidence_thresh=None):
     return fol.Keypoints(keypoints=keypoints)
 
 
-class FiftyOneYOLOConfig(Config):
+class FiftyOneYOLOConfig(fout.TorchImageModelConfig, fozm.HasZooModel):
```

Comment:
Inheriting from `HasZooModel` is good here, because it gives you a `download_model_if_necessary()` method that you can call from within your `Model` class to ensure that:
- the model is downloaded
- a `config.model_path` property is populated pointing to the model weights

I believe you're inheriting from `TorchImageModelConfig` so that you have access to the `entrypoint_fcn` and `entrypoint_args` parameters that it defines. However, this is a bit misleading because these models are not directly `torch.nn.Module`s and therefore many of the parameters defined by `TorchImageModelConfig` don't apply to these Ultralytics models.

Instead, it should be straightforward to delete the `load_yolo_model()` function you added and instead put that logic inside the `FiftyOneYOLOModel` class. The flow inside `FiftyOneYOLOModel`'s constructor will just be:
- call `config.download_model_if_necessary()` to populate 
- call `ultralytics.YOLO(config.model_path)`

Reply:
Took your advice. The one thing I'm not entirely satisfied with here is the `model_name` is not passing through properly, just the `model_path`. This is sufficient to load the appropriate checkpoint, but it would be nice to keep this additional information intact. Maybe there's something I'm doing wrong 🤷